pkg/storage/tsdb/caching_config.go

```diff
@@ -57,6 +58,22 @@ func (cfg *ChunksCacheConfig) Validate() error {
 	return cfg.BackendConfig.Validate()
 }
 
+func NewChunksCache(cfg ChunksCacheConfig, logger log.Logger, reg prometheus.Registerer) (chunkscache.ChunksCache, error) {
+	switch cfg.Backend {
+	case "":
+		return nil, nil
+	case cache.BackendMemcached:
+		client, err := cache.NewMemcachedClientWithConfig(logger, "chunks-cache", cfg.Memcached.ToMemcachedClientConfig(), prometheus.WrapRegistererWithPrefix("thanos_", reg))
```

Comment:
I think this conflicts with the same chunks cache name created by `CreateCachingBucket()`.